lib/TopSitesFeed.jsm

```diff
@@ -198,24 +208,43 @@ this.TopSitesFeed = class TopSitesFeed {
       return copy;
     }));
 
-    // Remove any duplicates from frecent and default sites
-    const [, dedupedFrecent, dedupedDefaults] = this.dedupe.group(
-      pinned, frecent, notBlockedDefaultSites);
-    const dedupedUnpinned = [...dedupedFrecent, ...dedupedDefaults];
-
-    // Remove adult sites if we need to
-    const checkedAdult = this.store.getState().Prefs.values.filterAdult ?
-      filterAdult(dedupedUnpinned) : dedupedUnpinned;
+    const dedupedUnpinned = this.dedupeTopSites({
+      // Convert any frecent site that matches TOPSITE_SEARCH_PROVIDERS to a search topsite
+      // without pinning or changing position
+      frecent: frecent.map(this.topSiteToSearchTopSite),
```

Comment:
I wonder if it would make more sense to move this to ~line 151 to be more consistent with how you are handling defaults and also the other search experiment?